homeassistant/components/sungrow/sensor.py

```diff
@@ -0,0 +1,436 @@
+    ),
+    SensorEntityDescription(
+        key="daily_running_time",
+        device_class=SensorDeviceClass.DURATION,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="country",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:earth",
+    ),
+    SensorEntityDescription(
+        key="monthly_power_yield", entity_registry_enabled_default=False
+    ),
+    SensorEntityDescription(
+        key="negative_voltage_to_ground",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="bus_voltage",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="power_factor_setting",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_adjustment",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_adjustment_switch",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+    SensorEntityDescription(
+        key="reactive_power_percentage",
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add solarlog entry."""
+    coordinator: SungrowData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SungrowSensor(
+            coordinator,
+            coordinator.battery_info
+            if description.key in BATTERY_DEVICE_VARIABLES
+            else coordinator.device_info,
+            description,
+        )
+        for description in DESCRIPTIONS
+        if description.key in coordinator.client.keys
+    )
+
+
+class SungrowSensor(SungrowCoordinatorEntity, SensorEntity):
+    """Representation of a Sensor."""
+
+    def __init__(
+        self,
+        coordinator: SungrowData,
+        device_info: DeviceInfo,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, device_info, description)
+        key = description.key
+
+        # set entity metadata based on variable metadata
+        variable = self.variable
+        self._attr_native_unit_of_measurement = variable.unit
+        if (
+            key.startswith("total_")
+            or key.startswith("daily_")
+            or key.startswith("monthly_")
+        ) and variable.unit == "kWh":
+            self._attr_state_class = SensorStateClass.TOTAL_INCREASING
```

Comment:
As you already use the SensorEnityDescription, please add the state class directly there.